routing/missioncontrol.go

```diff
@@ -124,6 +129,10 @@ type MissionControlConfig struct {
 	// Estimator gives probability estimates for node pairs.
 	Estimator Estimator
 
+	// OnConfigUpdate is function that is called whenever the
+	// mission control state is updated.
+	OnConfigUpdate func(cfg *MissionControlConfig)
+
```

Comment:
this also needs to be `fn.Option[func(cfg *MissionControlConfig)]` though.

Cause currently, if this is nil (which it is right now) then below when you do `onConfigUpdate: fn.Some(cfg.OnConfigUpdate),`, you are setting the callback to a nil. meaning that
```
m.onConfigUpdate.WhenSome(func(f func(cfg *MissionControlConfig)) {
		f(cfg)
	})
``` 
will be executed & cause a panic. 

You can see this by running this commit & calling `lncli setmccfg ...` -> it will cause LND to panic & crash

Reply:
Sorry, I thought `fn` would take care of it (it is new to me) ... but I should test it.